RecoHGCal/TICL/interface/SeedingRegionAlgoBase.h

```diff
@@ -0,0 +1,38 @@
+// Authors: Felice Pantaleo, Marco Rovere
+// Emails: felice.pantaleo@cern.ch, marco.rovere@cern.ch
+// Date: 06/2019
+
+#ifndef RecoHGCal_TICL_SeedingRegionAlgoBase_H__
+#define RecoHGCal_TICL_SeedingRegionAlgoBase_H__
+
+#include <memory>
+#include <vector>
+#include "DataFormats/HGCalReco/interface/Common.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "DataFormats/HGCalReco/interface/TICLSeedingRegion.h"
+#include "FWCore/Framework/interface/ConsumesCollector.h"
+
+namespace edm {
+  class Event;
+  class EventSetup;
+}  // namespace edm
+
+namespace ticl {
+  class SeedingRegionAlgoBase {
+  public:
+    SeedingRegionAlgoBase(const edm::ParameterSet& conf, edm::ConsumesCollector& sumes)
+        : algo_verbosity_(conf.getParameter<int>("algo_verbosity")) {}
+    virtual ~SeedingRegionAlgoBase(){};
+
+    virtual void makeRegions(const edm::Event& ev,
+                             const edm::EventSetup& es,
+                             std::vector<ticl::TICLSeedingRegion>& result) = 0;
+    enum VerbosityLevel { None = 0, Basic, Advanced, Expert, Guru };
```

Comment:
better to use enum class

Reply:
Hi @kpedro88 
I see the same enum VerbosityLevel { None = 0, Basic, Advanced, Expert, Guru }; 
syntax in RecoHGCal/TICL/interface/PatternRecognitionAlgoBase.h
why should it be different for  SeedingRegionAlgoBase.h? Or can you clarify better what you mean? thanks

Reply:
I should have made the same comment on `PatternRecognitionAlgoBase`, but must have missed it.

It would be even better not to duplicate this enum and just reference it from a central header (or something).

Reply:
Hi @kpedro88 
I looked for enum class, but then the use of Verbosity level as int like in https://github.com/cms-sw/cmssw/blob/f30411b0af6599cb68961bccbc04b9b42f0f9d4c/RecoHGCal/TICL/plugins/PatternRecognitionbyCA.cc#L36
is no more allowed.
To keep this nice feature, do you have any suggestion?
also, @rovere

Reply:
The type of the `algo_verbosity_` variable should be changed to the enum class type, rather than int. This is actually a benefit, because it prevents improper comparisons with other int variables whose values do not actually come from the enum.

Reply:
I do not see a huge added value, especially in this debugging only case.
@amartelli you could implement the comparison operator, templated, to accommodate comparison among enums and ints. something along the lines:

```c++
template<typename T>
bool operator>(const VerbosityLevel  lhs, const T  rhs) {return static_cast<int>(lhs) > static_cast<int>(rhs);}
```

which will somehow defeat the whole purpose :) 
Use the non-templated version, i.e. using `VerbosityLevel` on both arguments explicitly to prevent compilation against int-comparison.